### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lefteris-L**
[DEMO LINK](https://lefteris-l.github.io/react_pagination/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -21,6 +45,11 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            value={perPage}
+            onChange={(event) => {
+              setPerPage(+event.target.value);
```
Comment (meforoma) [inline src/App.tsx:50]: use explicit type conversion
```suggestion
              setPerPage(Number(event.target.value));
```

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1 +1,94 @@
...
+          aria-disabled={isFirstPage}
+          onClick={() => {
+            if (!isFirstPage) {
+              onPageChange(prev);
+            }
+          }}
+        >
+          «
+        </a>
+      </li>
```
Comment (meforoma) [inline src/components/Pagination/Pagination.tsx:51]: prev and next pages are 95% the same - make a tiny function that will render the proper button
like this:
```
const renderStepButton = (direction: 'next' | 'prev') => {...}
```
and use this short function instead two bulky similar constructs
`{renderStepButton('next')}` / `{renderStepButton('prev')}`

**Review by meforoma — CHANGES_REQUESTED** [review]
please see few suggestions
overall - good job 👏🏼

File: src/components/Pagination/Pagination.tsx (hunk 2 above)
Comment (lefteris-L) [reply src/components/Pagination/Pagination.tsx:51]: That sounds pretty cool. Thank you Roman

File: src/App.tsx [hunk 3]
```diff
@@ -1,18 +1,42 @@
...
+import './App.css';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
-export const App: React.FC = () => {
+export const App: FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
```
Comment (naziamoff) [inline src/App.tsx:12]: per page what? 😄

File: src/App.tsx [hunk 4]
```diff
@@ -1,18 +1,42 @@
...
+
+  const [
+    visibleItems,
+    setVisibleItems,
+  ] = useState(items.slice(0, perPage));
```
Comment (naziamoff) [inline src/App.tsx:17]: calculating the initial state value inside the useState() looks ugly a bit 🥲 
I'd create a variable for that

File: src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -1 +1,94 @@
...
+
+  const [prev, setPrev] = useState(currentPage - 1);
+  const [next, setNext] = useState(currentPage + 1);
+
+  useEffect(() => {
+    setNext(currentPage + 1);
+    setPrev(currentPage - 1);
+  }, [currentPage]);
+
+  const renderStepButton = (direction: 'prev' | 'next') => {
```
Comment (naziamoff) [inline src/components/Pagination/Pagination.tsx:28]: - better to separate the type. If you'll need to change it - you will need to look for exactly this piece of code
- if your values type can only be strings - it is much better to use Enum for that. If it is used the way you have it now - the possibility of making a mistake is much higher

File: src/App.tsx (hunk 3 above)
Comment (naziamoff) [inline src/App.tsx:12]: 5 looks kinda hardcoded here. It is better to create a variable for that to make it clear

**Review by naziamoff — CHANGES_REQUESTED** [review]
not bad at all. There's a space to improve though, so let's do it)

**Review by naziamoff — APPROVED** [review]
wow, great job!

**Pull request closed without merging**